### ISSUE TITLE
Add missing gradients

### ISSUE THREAD

**Issue by seanmor5** [elixir-nx/nx#161]
Currently missing gradients for these functions:

- [x] Product
- [ ] Scatter Window Max
- [ ] Scatter Window Min
- [x] Scatter Add
- [x] LU Decomposition
- [x] Triangular Solve
- [x] SVD

**Comment by polvalente** [comment]
https://github.com/tensorflow/tensorflow/pull/45475
https://github.com/tensorflow/tensorflow/issues/6504

This issue and Pull Request are related to QR grad. Might be relevant for who tackles this later.
In the PR comments there's a paper related to backpropagation of QR and LQ factorizations.

**Comment by Benjamin-Philip** [comment]
I believe that gradients for some of these functions have been implemented. Could you update the list @seanmor5 ?

**Comment by josevalim** [comment]
@Benjamin-Philip i recently updated the list and it should be good. If anything is missing, please let me know :)

**Comment by Benjamin-Philip** [comment]
@josevalim, you're right! I remember seeing some completed gradients a few days ago. I did not realise that it was updated. My bad 🙂.

**Comment by polvalente** [comment]
I'm starting work on triangular solve gradient
